ocaml/utils/language_extension.mli

```diff
@@ -1,53 +1,81 @@
 (** Language extensions provided by ocaml-jst *)
 
+(** A setting for extensions that track multiple maturity levels *)
 type maturity = Stable | Beta | Alpha
 
-(** The type of language extensions *)
-type t =
-  | Comprehensions
-  | Local
-  | Include_functor
-  | Polymorphic_parameters
-  | Immutable_arrays
-  | Module_strengthening
-  | Layouts of maturity
+(** The type of language extensions. An ['a t] is an extension that can either
+    be off or be set to have any value in ['a], so a [unit t] can be either on
+    or off, while a [maturity t] can have different maturity settings. *)
+type _ t =
+  | Comprehensions : unit t
+  | Local : unit t
+  | Include_functor : unit t
+  | Polymorphic_parameters : unit t
+  | Immutable_arrays : unit t
+  | Module_strengthening : unit t
+  | Layouts : maturity t
+
+(** Existentially packed language extension *)
+module Exist : sig
+  type 'a extn := 'a t
+  type t =
+    | Pack : 'a extn -> t
+
+  val to_string : t -> string
+  val is_enabled : t -> bool
+  val is_erasable : t -> bool
+
+  (** Returns a list of all strings, like ["layouts_beta"], that
+      correspond to this extension. *)
+  val to_command_line_strings : t -> string list
+
+  val all : t list
+end
 
 (** Equality on language extensions *)
-val equal : t -> t -> bool
+val equal : 'a t -> 'b t -> bool
 
-(** A list of all possible language extensions *)
-val all : t list
+(** Disable all extensions *)
+val disable_all : unit -> unit
 
-(** A maximal list of compatible language extensions (of the layouts extensions,
-    "layouts_alpha" is selected). *)
-val max_compatible : t list
+(** Maximally enable all extensions (that is, set to [Alpha] for [maturity]
+    extensions. *)
+val enable_maximal : unit -> unit
 
 (** Check if a language extension is "erasable", i.e. whether it can be
     harmlessly translated to attributes and compiled with the upstream
     compiler. *)
-val is_erasable : t -> bool
+val is_erasable : 'a t -> bool
 
 (** Print and parse language extensions; parsing is case-insensitive *)
-val to_string : t -> string
-val of_string : string -> t option
-val of_string_exn : string -> t
+val to_string : 'a t -> string
+val of_string : string -> Exist.t option
+
+(** Enable and disable according to command-line strings; these raise
+    an exception if the input string is invalid. *)
+val enable_of_string_exn : string -> unit
+val disable_of_string_exn : string -> unit
```

Comment:
Is it expected that enabling `layouts` but disabling `layouts_alpha` will result in `layouts` being disabled?

Reply:
Yes. I ran this change in semantics by Chris and Carl.